### PR TITLE
Fix a bug in the grammar of @supports

### PR DESCRIPTION
**Pull request by esdoppio**
in the commit message of 698e3c4

### CODE REVIEW

File: grammars/scss.cson [hunk 1]
```diff
@@ -918,9 +908,21 @@
...
+        'include': '#logical_operators'
+      }
+      {
+        'include': '#comparison_operators'
+      }
+    ]
   'logical_operators':
-    'match': '\\b(==|!=|<=|>=|<|>|not|or|and)\\b'
-    'name': 'keyword.control.operator'
+    'match': '\\b(?:not\\b|or\\b|and\\b)'
```
Comment (50Wliu) [inline grammars/scss.cson]: `\\b(not|or|and)\\b`

**Comment by 50Wliu** [comment]
Please rebase to get rid of the ellipse commit.

File: grammars/scss.cson (hunk 1 above)
Comment (esdoppio) [inline grammars/scss.cson]: is `\\b(not\\b|or\\b|and\\b)` ok?

Comment (50Wliu) [inline grammars/scss.cson]: Yes, why?

Comment (esdoppio) [inline grammars/scss.cson]: Performance. 😆

Comment (esdoppio) [inline grammars/scss.cson]: Performance. :laughing:

**Comment by esdoppio** [comment]
Anything else?

**Comment by 50Wliu** [comment]
Just to increase my understanding of things: how does this fix the referenced issue?  Is it because `#properties` puts things in the correct order, whereas the current manual order did it wrong?

**Comment by esdoppio** [comment]
First, the problem is actually I overlooked the fact that some word can be both a property and a value in CSS—`flex` in this case (can't say there won't be more in the future)—in the old way, it will always be tokenized as either a property or value, depending on, `#property_names` and `#property_values`, which comes first in the grammar.

```cson
{
  'include': '#constant_property_value'
}
{
  'include': '#property_names'
}
{
  'include': '#property_values'
}
```

In the old way, at first, there were just `#property_names` and `#property_values` (`flex` always tokenized as property name), I patched it by placing a `#constant_property_value` at the beginning (now always value) to make the test pass. But as you can see in the new specs, this didn't really solve the problem.

The fix mainly comes from that `#properties` wraps a layer around `#property_values` which specifies when exactly a token can be a value (begin and end):

```cson
{
  'begin': '(:)\\s*(?!(\\s*{))'
  …
  'end': '\\s*(;|(?=}|\\)))'
  …
  'contentName': 'meta.property-value.scss'
  'patterns': [
    …
    {
      'include': '#property_values'
    }
  ]
}
```

**Comment by 50Wliu** [comment]
Ok, that's what I was thinking, but just wanted to double-check.  Thanks.

**Comment by 50Wliu** [comment]
(Wrong button :see_no_evil:)

**Comment by esdoppio** [comment]
No problem 😆

**Pull request merged**